include/xtensor/xbroadcast.hpp

```diff
@@ -183,6 +214,11 @@ namespace xt
 
         CT m_e;
         inner_shape_type m_shape;
+
+        friend constexpr auto ::xt::detail::linear_begin<>(self_type&) noexcept;
+        friend constexpr auto ::xt::detail::linear_end<>(self_type&) noexcept;
+        friend constexpr auto ::xt::detail::linear_begin<>(const self_type&) noexcept;
+        friend constexpr auto ::xt::detail::linear_end<>(const self_type&) noexcept;
```

Comment:
`constexpr` forbidden in friend declarations refering to a specialization of a template, see [http://eel.is/c%2B%2Bdraft/temp.friend#8](http://eel.is/c%2B%2Bdraft/temp.friend#8) 

A first solution is to expose access methods to `e` (although this breaks encapsulation), the other one is to add a friend class in the detail namespace which exposes these accessors. Although strictly speaking this also breaks encapsulation, it won't be part of the API and people won't be tempted to use them.

Reply:
fixed. there was already an `expression()` doing exactly that.